keyvi/include/keyvi/dictionary/fsa/generator_adapter.h

```diff
@@ -89,11 +90,10 @@ class GeneratorAdapter final : public GeneratorAdapterInterface<PersistenceT, Va
   Generator<PersistenceT, ValueStoreT, OffsetTypeT, HashCodeTypeT> generator_;
 };
 
+template <typename ValueT>
 template <class PersistenceT, class ValueStoreT>
-typename GeneratorAdapterInterface<PersistenceT, ValueStoreT>::AdapterPtr
-GeneratorAdapterInterface<PersistenceT, ValueStoreT>::CreateGenerator(size_t size_of_keys,
-                                                                      const keyvi::util::parameters_t& params,
-                                                                      ValueStoreT* value_store) {
+typename GeneratorAdapterInterface<ValueT>::AdapterPtr GeneratorAdapterInterface<ValueT>::CreateGenerator(
```

Comment:
Seems that we are passing 2 value store related template params, maybe we can eliminate one of them ?

Reply:
this looks strange but note that ValueT is the template of the class, PersistenceT, ValueStoreT are templates of the method.